File: challenges/job-queue/python/starter/exercise.py
"""Implement the contract in the challenge README. Reference code is separate."""
from tooling.python.contracts import Clock, Delivery


class JobQueue:

    def __init__(self, visibility_ms: int, max_attempts: int, clock: Clock):
        raise NotImplementedError('Not implemented: __init__')

    def enqueue(self, payload: str) -> str:
        raise NotImplementedError('Not implemented: enqueue')

    def reserve(self) -> Delivery | None:
        raise NotImplementedError('Not implemented: reserve')

    def ack(self, receipt: str) -> bool:
        raise NotImplementedError('Not implemented: ack')

    def nack(self, receipt: str) -> bool:
        raise NotImplementedError('Not implemented: nack')

    def dead_letters(self) -> list[dict]:
        raise NotImplementedError('Not implemented: dead_letters')

File: challenges/job-queue/go/starter/exercise.go
package exercise

import "refresher/tooling/go/contracts"

// Add the state required by the challenge contract.
type JobQueue struct{}

func NewJobQueue(visibilityMS int64, maxAttempts int, clock contracts.Clock) *JobQueue {
	panic("Not implemented")
}

func (q *JobQueue) Enqueue(payload string) string {
	panic("Not implemented")
}

func (q *JobQueue) Reserve() *contracts.Delivery {
	panic("Not implemented")
}

func (q *JobQueue) Ack(receipt string) bool {
	panic("Not implemented")
}

func (q *JobQueue) Nack(receipt string) bool {
	panic("Not implemented")
}

func (q *JobQueue) DeadLetters() []contracts.DeadLetter {
	panic("Not implemented")
}
